File: python-selenium-pytest-bdd/src/saucedemo_uitests/pages/checkout_complete_page.py
from selenium.webdriver.common.by import By

from saucedemo_uitests.pages.base_page import BasePage, Locator

_COMPLETE_HEADER: Locator = (By.CSS_SELECTOR, "[data-test='complete-header']")


class CheckoutCompletePage(BasePage):
    def get_confirmation_message(self) -> str:
        return self.text_of(_COMPLETE_HEADER)

File: ts-playwright-test/src/pages/checkoutCompletePage.ts
import type { Locator, Page } from '@playwright/test';
import { BasePage } from './basePage';
import type { TestLogger } from '../support/testLogger';

export class CheckoutCompletePage extends BasePage {
  readonly confirmationHeader: Locator;

  constructor(page: Page, log: TestLogger) {
    super(page, log);
    this.confirmationHeader = page.getByTestId('complete-header');
  }
}
